webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -835,28 +835,28 @@ trait PromiseGeneration_BuildNodeContext {
       val all = nodeIds.foldLeft(NodesContextResult(Map(), Map())) { case (res, nodeId) =>
         (for {
           info         <- Box(allNodeInfos.get(nodeId)) ?~! s"Node with ID ${nodeId.value} was not found"
-          nodeTargets  =  allGroups.getTarget(info).map(_._2).toList
-          timeMerge    =  System.nanoTime
-          nodeInfo     <- MergeNodeProperties.withGroups(info, nodeTargets).toBox
-          _            =  {timeNanoMergeProp = timeNanoMergeProp + System.nanoTime - timeMerge}
-          policyServer <- Box(allNodeInfos.get(nodeInfo.policyServerId)) ?~! s"Node with ID ${nodeId.value} was not found"
-          nodeContext  <- systemVarService.getSystemVariables(nodeInfo, allNodeInfos, nodeTargets, globalSystemVariables, globalAgentRun, globalComplianceMode  : ComplianceMode)
+          policyServer <- Box(allNodeInfos.get(info.policyServerId)) ?~! s"Node with ID ${nodeId.value} was not found"
           context      =  ParamInterpolationContext(
-                              nodeInfo
+                              info
                             , policyServer
                             , globalPolicyMode
-                            , nodeContext
                             , parameters
                           )
-          nodeParam   <- Control.bestEffort(parameters.toSeq) { case (name, param) =>
+          nodeParam   <- parameters.toList.traverse { case (name, param) =>
                             for {
-                              p <- param(context).toBox
+                              p <- param(context)
                             } yield {
                               (name, p)
                             }
-                          }
+                          }.toBox
+          nodeTargets  =  allGroups.getTarget(info).map(_._2).toList
+          timeMerge    =  System.nanoTime
+          mergedProps  <- MergeNodeProperties.withDefaults(info, nodeTargets, nodeParam.map{case(k,v) => (k, GenericPropertyUtils.parseValue(v))}.toMap).toBox
+          _            =  {timeNanoMergeProp = timeNanoMergeProp + System.nanoTime - timeMerge}
+          nodeInfo     =  info.modify(_.node.properties).setTo(mergedProps.map(_.prop))
```

Comment:
funny because it now looks almost like previous code at least in my memory ;) Thanks to the new overridingService and the property hierarchy, everything is resolved at node level